docs/guides/es2k/building-p4cp-acc.md

```diff
@@ -0,0 +1,132 @@
+
+See [Building Stratum Dependencies for the ACC](../../../setup/building-es2k-stratum-deps.md)
+for instructions.
+
+You will need the Host depencies, Target dependencies, and Build environment
+setup file, in order to cross-compile P4 Control Plane for the ACC.
+
+## Fetch Source Code
+
+If you have not already done so, you will need to get a copy of the IPDK
+networking-recipe repository and its submodules:
+
+```bash
+git clone --recursive https://github.com/ipdk-io/networking-recipe.git ipdk.recipe
+```
+
+You may substitute your own local directory name for `ipdk.recipe`.
+
+## Build with OVS
+
+### Setup
+
+First, change to the source directory:
+
+```bash
+cd ipdk.recipe
+```
+
+If you have not already done so, source the file that the defines the
+[target build environment variables](../../../setup/building-es2k-stratum-deps.md#5-defining-the-target-build-environment).
+For example:
+
+```bash
+source setup/es2k-setup.env
+```
+
+### Building OVS
+
+The distribution includes a helper script (`make-cross-ovs.sh`) that can be
+used to build OVS for P4 Control Plane.
+
+> The `--help` (`-h`) option lists the parameters the helper script supports.
+
+To build OVS and install it in the sysroot directory under `/opt/ipdk/ovs`:
+
+```bash
+./make-cross-ovs.sh --prefix=//opt/ipdk/ovs
+```
+
+Options:
+
+- `--prefix=PREFIX` - where to install OVS
+
+The `//` at the beginning of the prefix path is a shortcut provided by
+the helper script. It will be replaced with the sysroot directory path.
+
+### Configuring P4 Control Plane
+
+The distribution includes a helper script (`config-cross-deps.sh`) that
+can be used to configure CMake to build P4 Control Plane.
+
+> The `--help` (`-h`) option lists the parameters the helper script supports.
+
+To configure CMake to build P4 Control Plane and install it in the sysroot
+directory under `/opt/ipdk/p4cp`:
+
+```bash
+./config-cross-recipe.sh \
+    --host=../hostdeps --deps=//opt/ipdk/deps \
+    --ovs=//opt/ipdk/ovs --sde=//opt/p4/p4sde \
+    --prefix=//opt/ipdk/p4cp
+```
+
+The options are:
+
+- `--host=HOST` - path to the Host dependencies
+- `--deps=DEPS` - path to the Target dependencies
+- `--ovs=OVS` - path to the OVS installation
+- `--sde=SDE` - path to the SDE installation
+- `--prefix=PREFIX` - where to install P4 control plane
+
+The `//` at the beginning of the prefix path is a shortcut provided by
+the helper script. It will be replaced with the sysroot directory path.
+
+### Building P4 Control Plane
+
+Now use CMake to build and install P4 Control Plane:
+
+```bash
+cmake --build build -j8 --target install
+```
+
+## Build without OVS
+
+To configure CMake to build P4 Control Plane without OVS:
+
+```bash
+./config-cross-recipe.sh \
+    --host=../hostdeps --deps=//opt/ipdk/deps \
+    --no-ovs --sde=//opt/p4/p4sde \
+    --prefix=//opt/ipdk/p4cp
+```
+
+The `--no-ovs` option excludes OVS support.
+
+Now perform the build:
+
+```bash
+cmake --build build -j8 --target install
+```
```

Comment:
Should we add a cleanup section?

Reply:
I've added instructions for doing a clean build.